x-pack/legacy/plugins/maps/public/layers/styles/vector/components/legend/style_property_legend_row.js

```diff
@@ -91,8 +92,8 @@ export class StylePropertyLegendRow extends Component {
     return (
```

Comment:
There's a small bug that the field formatter is not being updated, when the underlying field formatter changes on field-change.

ie. `_fieldFormatter` should be updated when `this.props.style.getField().getName()` changes. The 'hadLoadedFieldFormatter` bool-flag should probably just be the field-name, and then check if it changes.

e.g.:
- add kibana_sample_logs documents
- dynamically style fill-color with `bytes`
- expand legend-row -> it formats a number (as expected)
- change dynamic field to `@timestamp`
- -> the legend values change (expected) but continue to be formatted as numbers (not expected)
- open and close the legend row -> date-range is formatted as date (as expected)

Reply:
I field this bug yesterday with regard to this problem. https://github.com/elastic/kibana/issues/52122. This should be fixed in a separate PR